### DISCUSSION TITLE
How to get the bound port if port is set to 0?

### DISCUSSION THREAD

**Discussion by johan-bjareholt** [pallets/quart#299] [Q&A] [answered]
So I have a use-case where we use quart for a test HTTP server for an application with a HTTP client.
We bind port 0 to get any available port, but also need to be able to get which port it actually bound to in the end.
We need to bind to port 0, since our tests can be run in parallell on a single machine, so just choosing a port and hoping that it is free is not possible.
We previously did the same thing with flask and were able to parse out the port from werkzeug.

**Comment by davidism** [comment, marked as answer]
Werkzeug's own tests use https://pypi.org/project/ephemeral-port-reserve/ to reserve a port to then listen on.

**Reply by johan-bjareholt** [reply]
That's a pretty clever trick, will try it out and get back to you if it works for our use-case.

**Reply by johan-bjareholt** [reply]
Thanks, it worked for us. We used portpicker instead but it pretty much does the same thing.
